src/marqo/tensor_search/tensor_search.py

```diff
@@ -999,8 +1008,11 @@ def _vector_text_search(
         # KeyError indicates we have received a non-successful result
         try:
             if "index.max_result_window" in response["responses"][0]["error"]["root_cause"][0]["reason"]:
-                raise errors.IllegalRequestedDocCount("Marqo-OS rejected the response due to too many requested results. "
-                                             "Try reducing the query's limit parameter") from e
+                raise errors.IllegalRequestedDocCount(
+                    "Marqo-OS rejected the response due to too many requested results. "
+                    "Try reducing the query's limit parameter") from e
+            elif 'parse_exception' in response["responses"][0]["error"]["root_cause"][0]["reason"]:
```

Comment:
Is this related?

Reply:
Better error message: on multinode Marqo-os, the error message for bad filter string is slightly different. This helps handle it.